notation.go

```diff
@@ -380,16 +380,15 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, ve
 					logger.Error("Got nil outcome. Expecting non-nil outcome on verification failure")
 					return err
 				}
-
-				if _, ok := outcome.Error.(ErrorUserMetadataVerificationFailed); ok {
-					verificationFailedErr = outcome.Error
-				}
-
+				outcome.Error = fmt.Errorf("failed to verify signature with digest %v, %w", sigManifestDesc.Digest, outcome.Error)
+				verificationOutcomes = append(verificationOutcomes, outcome)
```

Comment:
The `Verify()` function states

> return the successful signature verification outcome.

However, this PR introduces a breaking change on the function behavior. It should be avoided.

Reply:
In order not to break, you may consider [`errors.Join()`](https://pkg.go.dev/errors#Join) to combine multiple errors into a single one.